storage/mongo.js

```diff
@@ -459,9 +460,15 @@ class MongoStore {
   writeMetricWithActorData(data, actorPerson) {
     try {
       if (actorPerson) {
-        data.actorEmail = actorPerson.emails[0];
-        data.actorDisplayName = actorPerson.displayName;
-        data.actorDomain = _.split(_.toLower(data.actorEmail), '@', 2)[1];
+        if (metricsStoreUserDetails) {
+          data.actorEmail = actorPerson.emails[0];
+          data.actorDisplayName = actorPerson.displayName;
+          data.actorDomain = _.split(_.toLower(data.actorEmail), '@', 2)[1];
+        }
+        else {
+          data.actorId = actorPerson.id;
+        }
+        
```

Comment:
I don't have any problem adding functionality to minimize the data stored in the metrics, but I'd prefer to see this implemented in a backwards compatible way so that existing apps that use the existing data in the metrics object don't need to change in order to keep their existing behavior.

Eg:
```js
     if (!metricsStoreActorIdOnly) {
        data.actorEmail = actorPerson.emails[0];
        data.actorDisplayName = actorPerson.displayName;	          
        data.actorDomain = _.split(_.toLower(data.actorEmail), '@', 2)[1];	          
      } else {
        data.actorId = actorPerson.id;
      }
    }